### PR TITLE
CP-10252: refactor navigation

### PR DESCRIPTION
**Pull request by ruijialin-avalabs**
## Description

**Ticket: [CP-10252] [CP-11944]**

Please provide:

- split screens into different protected routes guarded by WalletState + isLocked state
- moved protected screens into RootNavigator component
- replaced `NavigationRedirect` logic with protected routes 

## Screenshots/Videos

Include relevant screenshots or screen recordings of iOS and Android.

1. timeout

https://github.com/user-attachments/assets/746c6287-e971-4151-944e-e360367a334f

2. double push notification bypass

https://github.com/user-attachments/assets/4cbc0a2b-d565-41ac-a9e5-8fc32fb4dea9

## Testing

**Dev Testing (if applicable)**

- Provide steps to test the happy path of your feature
- Provide steps to test edge cases and error states
- Trigger a build on bitrise and reference it here
- Move the ticket into the "Testing" column on Jira

1. screen timeout should also lock the app and not skip pin screen
    - set your device screen timeout 15 seconds 
    - open the core mobile and leave it for 1 min 
    - unlock the device and the core mobile shows upfront since it’s the last app you opened 
    - enter pin screen shows up briefly and the app shows the portfolio (on Android, the last screen still present before the lock screen shows, this is due to in native Android, some Android devices (e.g. pixel devices) do not insert privacy window, and on js side, android could not insert custom logoModal on time before app goes into background.
 
4. iOS: background the app & require PIN immediately ON & user FACe ID OFF > send notification and deeplink > bypass the biometrics and enter the portfoilo without the auth. 

5. double push notification bypass pin screen
iOS only scenario: 
    - kill the app and try to deeplink 
    - land on the enter pin screen, dismiss the biometrics, and put the app in the background 
    - send another push notification and try to deeplink it 
    - verify you bypass the enter pin screen 
Android behavior: 
    - We don’t bypass the enter pin screen with the steps above, but we show a broken ui 

**QA Testing (if applicable)**

- Provide instructions for QA to test this feature thoroughly
- State expected behavior / acceptance criteria

## Checklist

Please check all that apply (if applicable)

- [X] I have performed a self-review of my code
- [X] I have verified the code works
- [X] I have included screenshots / videos of android and ios
- [X] I have added testing steps
- [ ] I have added/updated necessary unit tests
- [ ] I have updated the documentation

[CP-10252]: https://ava-labs.atlassian.net/browse/CP-10252?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CP-11944]: https://ava-labs.atlassian.net/browse/CP-11944?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by B0Y3R-AVA — DISMISSED** [review]

File: packages/core-mobile/app/contexts/WalletStateContext.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+// This hook can be used to access the user info.
+export const useWalletState = (): {
+  isLocked: boolean
+  isPrivacyOn: boolean
+  walletState: WalletState
+} => {
+  const value = use(WalletStateContext)
+  if (!value) {
+    throw new Error('useWalletState must be wrapped in a <AuthProvider />')
```
Comment (atn4z7) [inline packages/core-mobile/app/contexts/WalletStateContext.tsx:29]: which AuthProvider?

File: packages/core-mobile/app/new/common/components/NavigationRedirect.tsx
Comment (atn4z7) [inline packages/core-mobile/app/new/common/components/NavigationRedirect.tsx:1]: nice removal!!!!

File: packages/core-mobile/app/new/routes/RootNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  return (
+    <Stack
+      screenOptions={{
+        ...stackNavigatorScreenOptions,
+        headerShown: false
+      }}>
+      {/* verified and wallet active */}
+      <Stack.Protected guard={!isLocked && walletState === WalletState.ACTIVE}>
+        <Stack.Screen
+          name="(signedIn)"
```
Comment (atn4z7) [inline packages/core-mobile/app/new/routes/RootNavigator.tsx:24]: is it necessary to check `isLocked` in this navigator? when isLocked is true, wallet state should be inactive and when it is false, wallet state should be active.

File: packages/core-mobile/app/new/routes/RootNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            ...modalScreensOptions,
+            gestureEnabled: false
+          }}
+        />
+      </Stack.Protected>
+
+      {/* not verified */}
+      <Stack.Protected guard={isLocked}>
+        {/* wallet inactive */}
+        <Stack.Protected guard={walletState === WalletState.INACTIVE}>
```
[the remark this answers is not in the document]
Comment (ruijialin-avalabs) [reply packages/core-mobile/app/new/routes/RootNavigator.tsx:43]: I will remove isLocked.  and only depend on walletState.  I added isLocked just to be safe, since we are updating isLocked in the same place we update walletState.

**Review by atn4z7 — APPROVED** [review]
dev tested. working nicely!
